src/couch/src/test_util.erl

```diff
@@ -336,3 +337,13 @@ sort_apps(Apps) ->
 
 weight_app(couch_log) -> {0.0, couch_log};
 weight_app(Else) -> {1.0, Else}.
+
+maybe_add_js_engine(Apps) ->
+    case lists:member(js_engine, Apps) of
+        false ->
+            Apps;
+        true ->
+            application:start(config),
```

Comment:
This would cause a warning defined [here](https://github.com/apache/couchdb/pull/3676/files#diff-1386116efada5b26bab45d122ba9e6ba799e42d5adfdb6ce0b38f5abdd142fd8R102). Also `config` app will be restarted. To prevent it you could remove `config` from list of Apps. Since you started it already.